postgis/run_postgres

```diff
@@ -1,42 +1,33 @@
 #!/usr/bin/with-contenv sh
 
-mkdir -p ${PGDATA}
-
-# taken care of in initialization
-#chown -R postgres:postgres ${PGDATA}
-
-# if data folder is empty, initialize db
-test "$(ls -A "$PGDATA" 2>/dev/null)" || s6-setuidgid postgres /usr/bin/initdb -D $PGDATA
-
-# add md5 auth option for all internal clients
-[ $(grep -cE "host(\s)+all(\s)+all(\s)+0.0.0.0/0(\s)+md5" $PGDATA/pg_hba.conf) -eq 1 ] || \
-  echo "host all all 0.0.0.0/0 md5" >> $PGDATA/pg_hba.conf
-
-# add a user, pass and a database owned by that user
-TMP_FILE=/tmp/pgsql.initialize.txt
-# also add postgis extensions to that db
-TMP_GIS_FILE=/tmp/pggis.initialize.txt
-if [ ! -f $TMP_FILE ]; then
-    # create user and db
-    PGSQL_USER=${PGSQL_USER:-"dbuser"}
-    PGSQL_PASS=${PGSQL_PASS:-"dbpass"}
-    PGSQL_DB=${PGSQL_DB:-"dbname"}
-    cat > $TMP_FILE << EOSQL
+mkdir -p $PGDATA
+
+# if data folder is empty
+if [ "$(find $PGDATA ! -path $PGDATA | wc -l)" == "0" ]; then
+
+    # initialize db
+    s6-setuidgid postgres pg_ctl initdb -w -D $PGDATA
+
+    # add a user, pass and a postgis-enabled database owned by that user
+    # get rid of IPv6 error by forcing it to listen only on localhost
+    s6-setuidgid postgres pg_ctl start -w -o "-h '127.0.0.1'"
+
+    # add a user, pass and a postgis-enabled database owned by that user
+    s6-setuidgid postgres psql -a << EOF
 CREATE USER $PGSQL_USER WITH SUPERUSER;
 ALTER USER $PGSQL_USER WITH PASSWORD '$PGSQL_PASS';
 CREATE DATABASE $PGSQL_DB OWNER $PGSQL_USER;
-EOSQL
-    cat > $TMP_GIS_FILE << EOSQL
+EOF
+
+    # create postgis extensions on that database
+    s6-setuidgid postgres psql -a -U $PGSQL_USER $PGSQL_DB << EOF
 CREATE EXTENSION postgis;
 CREATE EXTENSION postgis_topology;
-EOSQL
-    s6-setuidgid postgres pg_ctl -w start
-    s6-setuidgid postgres psql -a -f $TMP_FILE
-    s6-setuidgid postgres psql -a -U $PGSQL_USER $PGSQL_DB -f $TMP_GIS_FILE
-    s6-setuidgid postgres pg_ctl stop
-fi
+EOF
 
-args="-D ${PGDATA}"
-[ -z ${PG_CONFIG_FILE} ] || args="--config-file=${PG_CONFIG_FILE}"
+    sleep 2
+    s6-setuidgid postgres pg_ctl -w stop
+
+fi
 
-exec s6-setuidgid postgres /usr/bin/postgres -h 0.0.0.0 $args
+exec s6-setuidgid postgres /usr/bin/postgres -h 0.0.0.0 -D ${PGDATA} --config-file=${PG_CONFIG_FILE}
```

Comment:
I'll believe you :-)

Reply:
proof:
https://github.com/UN-OCHA/docker-images/blob/998985cb4635e3deb8c0ebf83a55a08fd6eafff6/postgis/Dockerfile.tmpl#L31
and
https://github.com/UN-OCHA/docker-images/blob/998985cb4635e3deb8c0ebf83a55a08fd6eafff6/postgis/Dockerfile.tmpl#L32